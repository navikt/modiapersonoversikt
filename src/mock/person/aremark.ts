import { Diskresjonskoder, TilrettelagtKommunikasjonsTyper } from '../../konstanter';
import { BostatusTyper, Kjønn, Person, Relasjonstype, SivilstandTyper } from '../../models/person/person';
import { SikkerhetstiltakTyper } from '../../models/sikkerhetstiltak';
import { Personadresse } from '../../models/personadresse';

export const gateAdresseAremark: Personadresse = {
    endringsinfo: {
        sistEndret: '2017-11-25T21:29:14.049Z',
        sistEndretAv: 'SKD'
    },
    gateadresse: {
        gatenavn: 'Islandsgate',
        husnummer: '49',
        postnummer: '7693',
        poststed: 'Hovedøya'
    }
};

export const postboksAdresseAremark: Personadresse = {
    endringsinfo: {
        sistEndret: '2018-04-10T09:06:58.256Z',
        sistEndretAv: 'SKD'
    },
    postboksadresse: {
        postboksnummer: '9698',
        postnummer: '7705',
        poststed: 'Færder fyr'
    }
};

export const matrikkelAdresseAremark: Personadresse = {
    endringsinfo: {
        sistEndret: '2018-04-10T09:06:58.256Z',
        sistEndretAv: 'SKD'
    },
    matrikkeladresse: {
        postnummer: '5468',
        poststed: 'Lillomarka'
    }
};

export const utlandsAdresseAremark: Personadresse = {
    endringsinfo: {
        sistEndret: '2018-04-10T09:06:58.256Z',
        sistEndretAv: 'SKD'
    },
    utlandsadresse: {
        landkode: {
            kodeRef: 'SYD',
            beskrivelse: 'Syden'
        },
        adresselinjer: ['Ut av landet', 'Rundt svingen']
    }
};

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: Kjønn.Mann,
    geografiskTilknytning: '0118',
    alder: 42,
    folkeregistrertAdresse: gateAdresseAremark,
    alternativAdresse: postboksAdresseAremark,
    bankkonto: {
        banknavn: 'Bien sparebank',
        kontonummer: '12345678911',
        sistEndret: '2005-12-12',
        sistEndretAv: 'z9910',
        landkode: {
            kodeRef: 'NOR',
            beskrivelse: 'Norge'
        },
        valuta: {
            beskrivelse: 'Spesidaler',
            kodeRef: 'SPD'
        }
    },
    sikkerhetstiltak: {
        sikkerhetstiltaksbeskrivelse:
            'Får verken møte opp på NAV-kontor eller ringe på telefon ' + 'grunnet truende oppførsel og språk.',
        sikkerhetstiltakskode: SikkerhetstiltakTyper.FysiskOgTelefonUtestengelse,
        periode: {
            fra: '2005-12-12',
            til: '2025-12-12'
        }
    },
    navn: {
        endringsinfo: {
            sistEndret: '2012-12-12',
            sistEndretAv: 'NAV'
        },
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN'
    },
    tilrettelagtKomunikasjonsListe: [
        {
            kodeRef: 'TOHJ',
            beskrivelse: TilrettelagtKommunikasjonsTyper.TOHJ
        }
    ],
    diskresjonskode: { kodeRef: Diskresjonskoder.FORTROLIG_ADRESSE, beskrivelse: 'Sperret adresse, fortrolig' },
    statsborgerskap: { kodeRef: 'NOR', beskrivelse: 'NORGE' },
    personstatus: {
        dødsdato: undefined,
        bostatus: {
            kodeRef: BostatusTyper.Utvandret,
            beskrivelse: BostatusTyper.Utvandret
        }
    },
    sivilstand: {
        kodeRef: SivilstandTyper.Gift,
        beskrivelse: 'Gift',
        fraOgMed: '2005-12-12'
    },
    familierelasjoner: [
        {
            rolle: Relasjonstype.Barn,
            tilPerson: {
                navn: null,
                diskresjonskode: {
                    kodeRef: Diskresjonskoder.FORTROLIG_ADRESSE,
                    beskrivelse: 'Sperret adresse, fortrolig'
                },
                personstatus: {}
            }
        },
        {
            rolle: Relasjonstype.Barn,
            tilPerson: {
                navn: null,
                diskresjonskode: {
                    kodeRef: Diskresjonskoder.STRENGT_FORTROLIG_ADRESSE,
                    beskrivelse: 'Sperret adresse, strengt fortrolig'
                },
                personstatus: {}
            }
        },
        {
            harSammeBosted: true,
            rolle: Relasjonstype.Barn,
            tilPerson: {
                fødselsnummer: '10117100000',
                navn: {
                    sammensatt: 'Siri Bellomnavn Aremark',
                    fornavn: 'Siri',
                    mellomnavn: 'Bellomnavn',
                    etternavn: 'Aremark'
                },
                personstatus: {},
                alder: 15
            }
        },
        {
            harSammeBosted: false,
            rolle: Relasjonstype.Barn,
            tilPerson: {
                fødselsnummer: '10117100100',
                navn: {
                    sammensatt: 'Ola Ballomnavn Aremark',
                    fornavn: 'Ola',
                    mellomnavn: 'Ballomnavn',
                    etternavn: 'Aremark'
                },
                personstatus: {},
                alderMåneder: 5
            }
        },
        {
            harSammeBosted: false,
            rolle: Relasjonstype.Ektefelle,
            tilPerson: {
                navn: null,
                diskresjonskode: {
                    kodeRef: Diskresjonskoder.STRENGT_FORTROLIG_ADRESSE,
                    beskrivelse: 'Sperret adresse, strengt fortrolig'
                },
                personstatus: {}
            }
        }
    ],
    kontaktinformasjon: {
        mobil: {
            sistEndret: '2014-06-21T18:44:39+02:00',
            identifikator: '99887766',
            retningsnummer: { kodeRef: '+47', beskrivelse: 'Norge' },
            sistEndretAv: '10108000398'
        }
    }
};
