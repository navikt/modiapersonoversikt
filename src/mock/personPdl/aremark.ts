import {
    Adresse,
    AdresseBeskyttelse,
    EgenAnsatt,
    Enhet,
    Kjonn,
    KodeBeskrivelse,
    Navn,
    Person,
    PersonStatus,
    Sikkerhetstiltak,
    SikkerhetstiltakType,
    Sivilstand,
    SivilstandType,
    Statsborgerskap,
    Telefon,
    TilrettelagtKommunikasjon
} from '../../models/personPdl/person';
import { getAremarkForeldreansvar } from './foreldreansvarMock';
import { getDeltBostedMock } from './deltBostedMock';
import * as faker from 'faker';
import { getMockDodsbo } from './dodsbo';
import navfaker from 'nav-faker/dist/index';
import { mockVergemal } from './vergemalMock';
import { DigitalKontaktinformasjon } from '../../models/personPdl/digitalKontaktinformasjon';
import { getBankKonto } from './bankkontoMock';
import { Fullmakt, FullmaktsRolle } from '../../models/personPdl/fullmakt';

export const navnAremark: Navn = {
    fornavn: 'TESTFAMILIEN',
    mellomnavn: null,
    etternavn: 'AREMARK'
};

export const kodeBeskrivelseKjonn: KodeBeskrivelse<Kjonn> = {
    kode: Kjonn.Mann,
    beskrivelse: 'Mann'
};

export const adresseAremark: Adresse = {
    linje1: 'Adresseveien 1, 0661 AREMARKSTAD',
    linje2: null,
    linje3: null
};

export const navEnhetAremark: Enhet = {
    id: '0000',
    navn: 'Nav-kontor'
};

export const statsborgerskapAremark: Statsborgerskap = {
    land: {
        kode: 'NOR',
        beskrivelse: 'Norsk statsborger'
    },
    gyldigFraOgMed: new Date('1980-10-10'),
    gyldigTilOgMed: null
};

export const adressebeskyttelseAremark: KodeBeskrivelse<AdresseBeskyttelse> = {
    kode: AdresseBeskyttelse.Ugradert,
    beskrivelse: 'Ingen'
};

export const sikkerhetstiltakAremark: Sikkerhetstiltak = {
    type: SikkerhetstiltakType.TelefoniskUtestengelse,
    gyldigFraOgMed: new Date('2005-02-13'),
    gyldigTilOgMed: new Date('2020-02-02')
};

export const personstatusAremark: KodeBeskrivelse<PersonStatus> = {
    kode: PersonStatus.Bosatt,
    beskrivelse: 'Bosatt i Norge'
};

export const sivilstandAremark: Sivilstand = {
    type: {
        kode: SivilstandType.Skilt,
        beskrivelse: 'Skilt'
    },
    gyldigFraOgMed: new Date('2010-03-06')
};

export const tilrettelagtKommunikasjonAremark: TilrettelagtKommunikasjon = {
    talesprak: [
        {
            kode: 'NO',
            beskrivelse: 'Norsk'
        }
    ],
    tegnsprak: [
        {
            kode: 'NO',
            beskrivelse: 'Norsk'
        }
    ]
};

export const telefonAremark: Telefon = {
    retningsnummer: {
        kode: '+47',
        beskrivelse: 'retningsnummer er norsk'
    },
    identifikator: '99009900',
    sistEndret: new Date('2017-10-10'),
    sistEndretAv: 'TESTFAMILIEN AREMARK',
    prioritet: 1
};

export const digitalKontaktInformasjonAremark: DigitalKontaktinformasjon = {
    personident: '10108000398',
    reservasjon: 'reservert',
    epostadresse: {
        value: null,
        sistOppdatert: null,
        sisVerifisert: new Date('2013-01-01')
    },
    mobiltelefonnummer: {
        value: null,
        sistOppdatert: null,
        sisVerifisert: null
    }
};

export const fullmaktAremark: Fullmakt = {
    motpartsPersonident: '123456789',
    motpartsPersonNavn: {
        fornavn: 'Navn',
        mellomnavn: null,
        etternavn: 'Navnesen'
    },
    motpartsRolle: FullmaktsRolle.Fullmektig,
    omraade: ['*'],
    gyldigFraOgMed: new Date('2019-01-01'),
    gyldigTilOgMed: new Date('2022-12-12')
};

export const aremark: Person = {
    fnr: '10108000398',
    navn: [navnAremark],
    kjonn: [kodeBeskrivelseKjonn],
    fodselsdato: [new Date('1980-10-10')],
    dodsdato: [],
    bostedAdresse: [adresseAremark],
    kontaktAdresse: [adresseAremark],
    navEnhet: navEnhetAremark,
    statsborgerskap: [statsborgerskapAremark],
    adressebeskyttelse: [adressebeskyttelseAremark],
    sikkerhetstiltak: [sikkerhetstiltakAremark],
    erEgenAnsatt: EgenAnsatt.Nei,
    personstatus: [personstatusAremark],
    sivilstand: [sivilstandAremark],
    foreldreansvar: [getAremarkForeldreansvar()],
    deltBosted: getDeltBostedMock(faker),
    dodsbo: getMockDodsbo(faker),
    fullmakt: [fullmaktAremark],
    vergemal: mockVergemal(navfaker.personIdentifikator.f√∏dselsnummer()),
    tilrettelagtKommunikasjon: tilrettelagtKommunikasjonAremark,
    telefonnummer: [telefonAremark],
    kontaktOgReservasjon: digitalKontaktInformasjonAremark,
    bankkonto: getBankKonto()
};
