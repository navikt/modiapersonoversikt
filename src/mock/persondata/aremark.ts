import {
    AdresseBeskyttelse,
    EgenAnsatt,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    FullmaktsRolle,
    Kjonn,
    LocalDate,
    LocalDateTime,
    Person,
    PersonStatus,
    SivilstandType,
    Skifteform
} from '../../app/personside/visittkort-v2/PersondataDomain';
import { harDiskresjonskode } from '../../app/personside/visittkort-v2/visittkort-utils';

export function hentBarnAremark(): ForelderBarnRelasjon[] {
    return [
        {
            ident: '12345678910',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: 13,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [
                {
                    kode: AdresseBeskyttelse.UGRADERT,
                    beskrivelse: 'UGRADERT'
                }
            ],
            harSammeAdresse: false,
            dodsdato: []
        },
        {
            ident: '12345678911',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: null,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [],
            harSammeAdresse: false,
            dodsdato: []
        }
    ];
}

export const aremark: Person = {
    fnr: '10108000398',
    navn: [
        {
            fornavn: 'AREMARK',
            mellomnavn: null,
            etternavn: 'TESTFAMILIEN'
        }
    ],
    kjonn: [
        {
            kode: Kjonn.M,
            beskrivelse: 'Mann'
        }
    ],
    fodselsdato: ['1980-10-10' as LocalDate],
    geografiskTilknytning: '0118',
    alder: 41,
    dodsdato: [],
    bostedAdresse: [
        {
            linje1: 'Islandsgate 49',
            linje2: '7693 Hovedøya',
            linje3: 'Norge',
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2010-02-20T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    kontaktAdresse: [
        {
            linje1: 'Aremarkveien 1',
            linje2: '0320 Aremark',
            linje3: null,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    oppholdsAdresse: [
        {
            linje1: 'Oppholdsadresse 1',
            linje2: '0000 AREMARK',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    navEnhet: {
        id: '0219',
        navn: 'NAV Bærum',
        publikumsmottak: [
            {
                besoksadresse: {
                    linje1: 'Adressevei 1',
                    linje2: '0000 AREMARK',
                    linje3: null,
                    sistEndret: null,
                    gyldighetsPeriode: null
                },
                apningstider: [
                    {
                        ukedag: 'Mandag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Tirsdag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Onsdag',
                        apningstid: '10.00 - 16.00'
                    }
                ]
            }
        ]
    },
    statsborgerskap: [
        {
            land: {
                kode: 'NOR',
                beskrivelse: 'NORGE'
            },
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    adressebeskyttelse: [
        {
            kode: AdresseBeskyttelse.KODE6,
            beskrivelse: 'Kode 6'
        }
    ],
    sikkerhetstiltak: [
        {
            type: 'TFUS',
            beskrivelse: 'Telefonisk utestengelse',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    erEgenAnsatt: EgenAnsatt.JA,
    personstatus: [
        {
            kode: PersonStatus.BOSATT,
            beskrivelse: 'Bosatt i Norge'
        }
    ],
    sivilstand: [
        {
            type: {
                kode: SivilstandType.GIFT,
                beskrivelse: 'Gift'
            },
            gyldigFraOgMed: '2005-12-12' as LocalDate,
            sivilstandRelasjon: {
                fnr: '12345555555',
                navn: [
                    {
                        fornavn: 'BAREMARK',
                        mellomnavn: null,
                        etternavn: 'TESTFAMILIEN'
                    }
                ],
                alder: 40,
                adressebeskyttelse: [
                    {
                        kode: AdresseBeskyttelse.UGRADERT,
                        beskrivelse: 'UGRADERT'
                    }
                ],
                harSammeAdresse: true
            }
        }
    ],
    foreldreansvar: hentBarnAremark().map((barn) => ({
        ansvar: 'felles',
        ansvarlig: null,
        ansvarsubject: {
            navn: harDiskresjonskode(barn.adressebeskyttelse) ? null : barn.navn.firstOrNull(),
            ident: barn.ident
        }
    })),
    deltBosted: [
        {
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            },
            adresse: {
                linje1: `Adresseveien 1`,
                linje2: '0000 Aremark',
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            }
        }
    ],
    dodsbo: [
        {
            adressat: {
                advokatSomAdressat: {
                    kontaktperson: {
                        fornavn: 'Advokat',
                        mellomnavn: null,
                        etternavn: 'Navnesen'
                    },
                    organisasjonsnavn: 'Advokatkontoret Aremark',
                    organisasjonsnummer: '01234567891011'
                },
                organisasjonSomAdressat: null,
                personSomAdressat: null
            },
            adresse: {
                linje1: 'Elgelia 20, 0000 Aremark',
                linje2: null,
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            },
            registrert: '2010-02-02' as LocalDate,
            skifteform: Skifteform.OFFENTLIG,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'Folkeregisteret'
            }
        }
    ],
    fullmakt: [
        {
            motpartsPersonident: '123456789',
            motpartsPersonNavn: {
                fornavn: 'Navn',
                mellomnavn: null,
                etternavn: 'Navnesen'
            },
            motpartsRolle: FullmaktsRolle.FULLMEKTIG,
            omrade: [
                {
                    kode: '*',
                    beskrivelse: ''
                }
            ],
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    vergemal: [
        {
            ident: '21042900076',
            navn: {
                fornavn: 'Simen',
                mellomnavn: null,
                etternavn: 'Solli'
            },
            vergesakstype: 'Voksen',
            omfang: 'Ivareta personens interesser innenfor det personlige og økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-10' as LocalDate
            }
        },
        {
            ident: '123456799',
            navn: {
                fornavn: 'Truls',
                mellomnavn: null,
                etternavn: 'Tøffel'
            },
            vergesakstype: 'Fremtidsfullmakt',
            omfang: 'Ivareta personens interesser innenfor det økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-10' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    tilrettelagtKommunikasjon: {
        talesprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ],
        tegnsprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ]
    },
    telefonnummer: [
        {
            retningsnummer: {
                kode: '+47',
                beskrivelse: 'Norge'
            },
            identifikator: '99009900',
            sistEndret: {
                ident: '11223344',
                tidspunkt: '2018-06-14T00:00:00' as LocalDateTime,
                system: 'BD06'
            },
            prioritet: -1
        }
    ],
    kontaktOgReservasjon: {
        personident: '10108000398',
        reservasjon: 'true',
        epostadresse: {
            value: 'epost@nav.no',
            sistOppdatert: null,
            sistVerifisert: '2013-01-01' as LocalDate
        },
        mobiltelefonnummer: {
            value: '12345678',
            sistOppdatert: '2021-11-02' as LocalDate,
            sistVerifisert: '2021-11-02' as LocalDate
        }
    },
    bankkonto: {
        kontonummer: '12345678900',
        banknavn: 'DNB ASA',
        sistEndret: {
            ident: '1010800 BD03',
            tidspunkt: '2006-03-15T00:00:00' as LocalDateTime,
            system: ''
        },
        bankkode: null,
        swift: 'DNBANOKKXXX',
        landkode: null,
        adresse: {
            linje1: 'Bankveien 1,',
            linje2: '0357 Bankestad',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: null
        },
        valuta: {
            kode: 'NOK',
            beskrivelse: 'Norske kroner'
        }
    },
    forelderBarnRelasjon: hentBarnAremark()
};
