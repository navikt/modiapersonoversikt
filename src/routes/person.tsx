import { Navigate, createFileRoute } from '@tanstack/react-router';
import LegacyAppContainer from 'src/app/LegacyAppContainer';
import PersonoversiktWrapper from 'src/app/personside/Personoversikt';

export const Route = createFileRoute('/person')({
    component: PersonLayout,
    notFoundComponent: () => <Navigate to="/person/oversikt" replace />
});

function PersonLayout() {
    return (
        <LegacyAppContainer>
            <PersonoversiktWrapper />
        </LegacyAppContainer>
    );
}
