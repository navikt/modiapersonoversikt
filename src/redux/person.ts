import { Person } from '../models/person';
import { Action, Dispatch } from 'redux';
import { FetchError, FetchSuccess, RestActions, STATUS } from './utils';
import { getPerson } from '../api/person-api';
import { aremark } from '../mock/person-mock';

export interface PersonState {
    data: Person;
    status: STATUS;
    error: string | null;
}

const reducerName = 'person/';

const actionNames = {
    PENDING: reducerName + STATUS.PENDING,
    OK: reducerName + STATUS.OK,
    ERROR: reducerName + STATUS.ERROR
};

const initialState: PersonState = {
    data: aremark,
    status: STATUS.NOT_STARTED,
    error: null
};

export function hentPerson(fodselsnummer: string) {
    return (dispatch: Dispatch<Action>) => {
        dispatch({ type: actionNames.PENDING});
        return getPerson(fodselsnummer)
            .then((person) => {
                dispatch({type: actionNames.OK, data: person});
                return person;
            })
            .catch((err) => {
                dispatch({type: actionNames.ERROR, error: err});
            });
    };
}

export default function reducer(state: PersonState = initialState, action: RestActions<Person>): PersonState {
    switch (action.type) {
        case actionNames.PENDING:
            return {...state, status: STATUS.PENDING };
        case actionNames.OK:
            return {...state, status: STATUS.OK, data: (<FetchSuccess<Person>> action).data};
        case actionNames.ERROR:
            return {...state, status: STATUS.ERROR, error: (<FetchError> action).error};
        default:
            return state;
    }
}