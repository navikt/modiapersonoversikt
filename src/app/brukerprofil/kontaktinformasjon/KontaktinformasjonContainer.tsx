import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import Undertittel from 'nav-frontend-typografi/lib/undertittel';

import { AppState } from '../../../redux/reducers';
import Innholdslaster from '../../../components/Innholdslaster';
import { Person } from '../../../models/person/person';
import { KodeverkResponse } from '../../../models/kodeverk';
import { hentRetningsnummere } from '../../../redux/restReducers/kodeverk/retningsnummereReducer';
import KontaktinformasjonForm from './KontaktinformasjonForm';
import { isNotStarted, Loaded, RestResource } from '../../../redux/restReducers/restResource';
import { AsyncDispatch } from '../../../redux/ThunkTypes';

interface DispatchProps {
    hentRetningsnummer: () => void;
}

interface StateProps {
    retningsnummerResource: RestResource<KodeverkResponse>;
}

interface OwnProps {
    person: Person;
}

type Props = OwnProps & DispatchProps & StateProps;

const NavKontaktinformasjonWrapper = styled.div`
    margin-top: 2em;
`;

class KontaktinformasjonFormContainer extends React.Component<Props> {
    constructor(props: Props) {
        super(props);
    }

    componentDidMount() {
        if (isNotStarted(this.props.retningsnummerResource)) {
            this.props.hentRetningsnummer();
        }
    }

    render() {
        return (
            <div>
                <Undertittel>Kontaktinformasjon</Undertittel>
                <Innholdslaster avhengigheter={[this.props.retningsnummerResource]}>
                    <NavKontaktinformasjonWrapper>
                        <KontaktinformasjonForm
                            person={this.props.person}
                            retningsnummerKodeverk={
                                (this.props.retningsnummerResource as Loaded<KodeverkResponse>).data
                            }
                        />
                    </NavKontaktinformasjonWrapper>
                </Innholdslaster>
            </div>
        );
    }
}

const mapStateToProps = (state: AppState, ownProps: OwnProps): StateProps & OwnProps => {
    return {
        retningsnummerResource: state.restResources.retningsnummer,
        person: ownProps.person
    };
};

function mapDispatchToProps(dispatch: AsyncDispatch): DispatchProps {
    return {
        hentRetningsnummer: () => dispatch(hentRetningsnummere())
    };
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(KontaktinformasjonFormContainer);
