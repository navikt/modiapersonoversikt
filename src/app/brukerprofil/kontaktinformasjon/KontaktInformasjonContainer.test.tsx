import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../test/Testprovider';
import { aremark } from '../../../mock/person/aremark';
import KontaktinformasjonContainer from './KontaktinformasjonContainer';

test('Viser form for endre kontaktinformasjon', () => {
    const endreKontaktInformasjon = renderer.create(
        <TestProvider>
            <KontaktinformasjonContainer person={aremark} />
        </TestProvider>
    );

    expect(endreKontaktInformasjon.toJSON()).toMatchSnapshot();
});
