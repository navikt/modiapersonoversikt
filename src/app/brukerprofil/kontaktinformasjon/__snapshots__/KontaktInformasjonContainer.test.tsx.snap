// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser form for endre kontaktinformasjon 1`] = `
.c6 {
  color: #78706a;
  margin-top: 3px;
}

.c6 .typo-etikett-liten {
  line-height: 1rem;
}

.c4 {
  margin-right: 2em;
}

.c5 {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c7 > * {
  margin: 0.5rem 0 0.5rem 1em;
}

.c1 {
  margin-top: 1em;
}

.c2 {
  margin-bottom: 1em;
}

.c0 {
  margin-top: 2em;
}

<div>
  <h2
    className="typo-undertittel"
  >
    Kontaktinformasjon
  </h2>
  <div
    className="c0"
  >
    <form
      onSubmit={[Function]}
    >
      <p
        className="typo-undertekst-bold"
      >
        Telefonnummer bruker ønsker å bli oppringt på
      </p>
      <div
        className="c1"
      >
        <div
          className="c2"
        >
          <p
            className="typo-ingress"
          >
            Mobiltelefon
          </p>
          <div
            className="c3"
          >
            <div
              className="c4"
            >
              <div
                className="skjemaelement"
              >
                <label
                  className="skjemaelement__label"
                  htmlFor="Helt tilfeldig ID"
                >
                  Landkode
                </label>
                <div
                  className="selectContainer input--m"
                >
                  <select
                    className="skjemaelement__input"
                    disabled={false}
                    id="Helt tilfeldig ID"
                    onChange={[Function]}
                    value="+47"
                  >
                    <option
                      disabled={true}
                      value=""
                    >
                      Velg retningsnummer
                    </option>
                    <option
                      value="+47"
                    >
                      Norge
                       (
                      +47
                      )
                    </option>
                    <option
                      value="+54"
                    >
                      Argentina
                       (
                      +54
                      )
                    </option>
                    <option
                      value="+44"
                    >
                      Storbritannia (UK)
                       (
                      +44
                      )
                    </option>
                    <option
                      value="+46"
                    >
                      Sverige
                       (
                      +46
                      )
                    </option>
                  </select>
                </div>
                <div
                  aria-live="assertive"
                  role="alert"
                />
              </div>
            </div>
            <div
              className="c5"
            >
              <div
                className="skjemaelement"
              >
                <label
                  className="skjemaelement__label"
                  htmlFor="Helt tilfeldig ID"
                >
                  Telefonnummer
                </label>
                <input
                  className="skjemaelement__input input--xxl"
                  id="Helt tilfeldig ID"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  type="text"
                  value="998 87 766"
                />
                <div
                  aria-live="assertive"
                  role="alert"
                />
              </div>
            </div>
          </div>
          <div
            className="c6"
          >
            <p
              className="typo-etikett-liten"
            >
              Endret 
              21.06.2014
               
              av bruker
            </p>
          </div>
        </div>
        <div
          className="c2"
        >
          <p
            className="typo-ingress"
          >
            Hjemmenummer
          </p>
          <div
            className="c3"
          >
            <div
              className="c4"
            >
              <div
                className="skjemaelement"
              >
                <label
                  className="skjemaelement__label"
                  htmlFor="Helt tilfeldig ID"
                >
                  Landkode
                </label>
                <div
                  className="selectContainer input--m"
                >
                  <select
                    className="skjemaelement__input"
                    disabled={false}
                    id="Helt tilfeldig ID"
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      disabled={true}
                      value=""
                    >
                      Velg retningsnummer
                    </option>
                    <option
                      value="+47"
                    >
                      Norge
                       (
                      +47
                      )
                    </option>
                    <option
                      value="+54"
                    >
                      Argentina
                       (
                      +54
                      )
                    </option>
                    <option
                      value="+44"
                    >
                      Storbritannia (UK)
                       (
                      +44
                      )
                    </option>
                    <option
                      value="+46"
                    >
                      Sverige
                       (
                      +46
                      )
                    </option>
                  </select>
                </div>
                <div
                  aria-live="assertive"
                  role="alert"
                />
              </div>
            </div>
            <div
              className="c5"
            >
              <div
                className="skjemaelement"
              >
                <label
                  className="skjemaelement__label"
                  htmlFor="Helt tilfeldig ID"
                >
                  Telefonnummer
                </label>
                <input
                  className="skjemaelement__input input--xxl"
                  id="Helt tilfeldig ID"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  type="text"
                  value=""
                />
                <div
                  aria-live="assertive"
                  role="alert"
                />
              </div>
            </div>
          </div>
        </div>
        <p
          className="typo-ingress"
        >
          Jobbnummer
        </p>
        <div
          className="c3"
        >
          <div
            className="c4"
          >
            <div
              className="skjemaelement"
            >
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Landkode
              </label>
              <div
                className="selectContainer input--m"
              >
                <select
                  className="skjemaelement__input"
                  disabled={false}
                  id="Helt tilfeldig ID"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    disabled={true}
                    value=""
                  >
                    Velg retningsnummer
                  </option>
                  <option
                    value="+47"
                  >
                    Norge
                     (
                    +47
                    )
                  </option>
                  <option
                    value="+54"
                  >
                    Argentina
                     (
                    +54
                    )
                  </option>
                  <option
                    value="+44"
                  >
                    Storbritannia (UK)
                     (
                    +44
                    )
                  </option>
                  <option
                    value="+46"
                  >
                    Sverige
                     (
                    +46
                    )
                  </option>
                </select>
              </div>
              <div
                aria-live="assertive"
                role="alert"
              />
            </div>
          </div>
          <div
            className="c5"
          >
            <div
              className="skjemaelement"
            >
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Telefonnummer
              </label>
              <input
                className="skjemaelement__input input--xxl"
                id="Helt tilfeldig ID"
                onChange={[Function]}
                onKeyPress={[Function]}
                type="text"
                value=""
              />
              <div
                aria-live="assertive"
                role="alert"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c7"
      >
        <button
          className="knapp knapp--standard knapp--disabled"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Avbryt
        </button>
        <button
          className="knapp knapp--hoved knapp--disabled"
          disabled={true}
          type="submit"
        >
          Endre telefonnummer
        </button>
      </div>
    </form>
  </div>
</div>
`;
